e2e_tests/integrations/test_huggingface.py

```diff
@@ -170,48 +173,213 @@ def test_run_access_without_callback_configured(self):
-            assert run.exists(f"{base_namespace}/train/after_training_metric")
+        time.sleep(SECONDS_TO_WAIT_FOR_UPDATE)
+        run = init_run(
+            run=run_id,
+            project=environment.project,
+            api_token=environment.user_token,
+            mode="read-only",
+        )
+        assert run.exists(f"{base_namespace}/train")
+        assert run.exists(f"{base_namespace}/train/metric1")
+        assert run.exists(f"{base_namespace}/train/another/metric")
+        assert run.exists(f"{base_namespace}/train/after_training_metric")
 
-            assert run[f"{base_namespace}/train/metric1"].fetch_last() == 123
-            assert run[f"{base_namespace}/train/another/metric"].fetch_last() == 0.2
-            assert run[f"{base_namespace}/train/after_training_metric"].fetch_last() == 2501
+        assert run[f"{base_namespace}/train/metric1"].fetch_last() == 123
+        assert run[f"{base_namespace}/train/another/metric"].fetch_last() == 0.2
+        assert run[f"{base_namespace}/train/after_training_metric"].fetch_last() == 2501
 
     def test_integration_version_is_logged(self, environment):
         with init_run(project=environment.project, api_token=environment.user_token) as run:
+            run_id = run["sys/id"].fetch()
             callback = NeptuneCallback(run=run)
             trainer = Trainer(**self._trainer_default_attributes, callbacks=[callback])
             trainer.train()
-            run.sync()
 
-            assert run.exists("source_code/integrations/transformers")
+        run = init_run(
+            run=run_id,
+            project=environment.project,
+            api_token=environment.user_token,
+            mode="read-only",
+        )
+        assert run.exists("source_code/integrations/transformers")
+
+    def test_non_monitoring_runs_creation(self, environment):
+        # given
+        common_tag = fake.nic_handle()
+        project = get_project(name=environment.project, api_token=environment.user_token)
+
+        # and
+        callback = NeptuneCallback(
+            project=environment.project, api_token=environment.user_token, tags=common_tag
+        )
+        trainer = Trainer(**self._trainer_default_attributes, callbacks=[callback])
+
+        # when
+        trainer.log({"metric1": 123})
+        time.sleep(SECONDS_TO_WAIT_FOR_UPDATE)
+
+        # then
+        runs = project.fetch_runs_table(tag=common_tag).to_rows()
+        assert len(runs) == 1
+        with pytest.raises(ValueError):
+            runs[0].get_attribute_value("monitoring/cpu")
+        assert runs[0].get_attribute_value("finetuning/train/metric1") == 123
+
+        # when
+        trainer.train()
+        time.sleep(SECONDS_TO_WAIT_FOR_UPDATE)
+
+        # then
+        runs = project.fetch_runs_table(tag=common_tag).to_rows()
+        assert len(runs) == 1
+        assert runs[0].get_attribute_value("monitoring/cpu") is not None
+
+        # when
+        trainer.log({"metric2": 234})
```

Comment:
Why wasn't second run created here since previous one was closed and `_run` in callback was set to `None` in `on_train_end`?

Reply:
The user should be able to add some custom metrics after training so `log` operation will reinitialize last training' run :wink: